core/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableCommand.scala

```diff
@@ -0,0 +1,247 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.commands
+
+// scalastyle:off import.ordering.noEmptyLine
+import java.io.FileNotFoundException
+
+import org.apache.spark.sql.delta.{DeltaErrors, DeltaTimeTravelSpec, Snapshot}
+import org.apache.spark.sql.delta.DeltaOperations.Clone
+import org.apache.spark.sql.delta.actions.{AddFile, Metadata, Protocol}
+import org.apache.spark.sql.delta.catalog.DeltaTableV2
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.sql.{Column, Dataset, Row, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.catalog.CatalogTable
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.{LongType, StructType}
+import org.apache.spark.util.{Clock, SerializableConfiguration, SystemClock}
+// scalastyle:on import.ordering.noEmptyLine
+
+/**
+ * Clones a Delta table to a new location with a new table id.
+ * The clone can be performed as a shallow clone (i.e. shallow = true),
+ * where we do not copy the files, but just point to them.
+ * If a table exists at the given targetPath, that table will be replaced.
+ *
+ * @param sourceTable is the table to be cloned
+ * @param targetIdent destination table identifier to clone to
+ * @param tablePropertyOverrides user-defined table properties that should override any properties
+ *                       with the same key from the source table
+ * @param targetPath the actual destination
+ */
+case class CloneTableCommand(
+    sourceTable: CloneSource,
+    targetIdent: TableIdentifier,
+    tablePropertyOverrides: Map[String, String],
+    targetPath: Path)
+  extends CloneTableBase(sourceTable, tablePropertyOverrides, targetPath) {
+
+  import CloneTableCommand._
+
+
+  /** Return the CLONE command output from the execution metrics */
+  override protected def getOutputSeq(operationMetrics: Map[String, Long]): Seq[Row] = {
+    Seq(Row(
+      operationMetrics.get(SOURCE_TABLE_SIZE),
+      operationMetrics.get(SOURCE_NUM_OF_FILES),
+      operationMetrics.get(NUM_REMOVED_FILES),
+      operationMetrics.get(NUM_COPIED_FILES),
+      operationMetrics.get(REMOVED_FILES_SIZE),
+      operationMetrics.get(COPIED_FILES_SIZE)
+    ))
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    if (!targetPath.isAbsolute) {
```

Comment:
@jackierwzhang, do you remember why we need to validate if it is an absolute path?